File: .config/firefox/user.js/make.js
#!/bin/node
"use strict";

const fs = require("fs");
const path = require("path");

const read_prefs = function(prefs, f) {
    const user_pref = function(key, value) {
        prefs.set(key, value);
    };
    eval("" + fs.readFileSync(f));
};

const FIREFOX_HOME = path.normalize(__dirname + "/..");

const prefs = new Map();
read_prefs(prefs, FIREFOX_HOME + "/user.js/ghacks.js");
read_prefs(prefs, FIREFOX_HOME + "/user.js/my.js");

prefs.forEach((value, key) => {
    if(value === null) return;
    switch(typeof(value)) {
    case "boolean" :
    case "number" :
        console.log(`user_pref("${key}", ${value});`);
        break;
    case "string" :
        console.log(`user_pref("${key}", "${value}");`);
        break;
    default :
        throw `unknown type: ${typeof(value)}`;
    }
});

File: .config/firefox/user.js/make.py
#!/bin/python

import os, subprocess;

prefs = dict()
def user_pref(key, value) :
    prefs[key] = value;

def eval_file(f) :
    exec(subprocess.check_output(f"cat {f} | gcc -E -P -", shell = True, text = True))

false = False
true = True
null = None
FIREFOX_HOME = os.path.normpath(os.path.dirname(os.path.realpath(__file__)) + "/..")

eval_file(FIREFOX_HOME + "/user.js/ghacks.js");
eval_file(FIREFOX_HOME + "/user.js/my.js");

for name in prefs :
    value = prefs[name]
    if value == None : continue
    if isinstance(value, bool) and value == False :
        print(f'user_pref("{name}", false);')
        continue
    if isinstance(value, bool) and value == True :
        print(f'user_pref("{name}", true);')
        continue
    if isinstance(value, int) :
        print(f'user_pref("{name}", {value});')
        continue
    if isinstance(value, str) :
        print(f'user_pref("{name}", "{value}");')
        continue
    raise Exception(f'unknown type: {type(value)}')
